Tests/Opc.Ua.Client.Tests/ContinuationPointInBatchTest.cs

```diff
@@ -0,0 +1,1398 @@
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Castle.Core.Logging;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Diagnostics.Tracing.Parsers.AspNet;
+using Microsoft.Diagnostics.Tracing.Parsers.MicrosoftWindowsTCPIP;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using Opc.Ua.Server;
+using Opc.Ua.Server.Tests;
+using Quickstarts.ReferenceServer;
+using Assert = NUnit.Framework.Legacy.ClassicAssert;
+
+
+namespace Opc.Ua.Client.Tests
+{
+    public class ManagedBrowseExpectedResultValues
+    {
+        public uint InputMaxNumberOfContinuationPoints { get; set; } = 0;
+        public uint InputMaxNumberOfReferencesPerNode { get; set; } = 0;
+        public int ExpectedNumberOfPasses { get; set; } = 0;
+        public List<int> ExpectedNumberOfBadNoCPSCs { get; set; }
+
+    }
+
+    public class ManagedBrowseTestDataProvider
+    {
+        public uint MaxNumberOfContinuationPoints { get; set; } = 0;
+        public uint MaxNumberOfReferencesPerNode { get; set; } = 0;
+        public int ExpectedNumberOfPasses { get; set; } = 0;
+        public List<int> ExpectedNumberOfBadNoCPSCs { get; set; }
+    }
+
+
+
+
+    internal class CPBatchTestMemoryWriter : TextWriter
+    {
+        private MemoryStream m_stream = new MemoryStream(64000);
+        private StreamWriter m_writer = null;
+
+        public override Encoding Encoding => Encoding.Default;
+        public CPBatchTestMemoryWriter()
+        {
+            m_writer = new StreamWriter(m_stream);
+            m_writer.AutoFlush = true;
+        }
+        public override void Write(char value)
+        {                        
+                m_writer.Write(value);                        
+        }
+
+        public List<String> getEntries()
+        {
+            m_stream.Position = 0;
+            List<string> entries = new List<string>();
+            using (var sr = new StreamReader(m_stream))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null) { entries.Add(line); }
+            }
+            // get entries closes the stream.
+            m_stream = new MemoryStream(64000);
+            m_writer = new StreamWriter(m_stream);
+            m_writer.AutoFlush = true;
+            return entries;
+        }
+
+        new public void Dispose()
```

Comment:
this dispose should not be needed?

Reply:
I deleted the method and it seems to work....